### PR TITLE
Upgraded grafana sdk and some Grafana :bug: fixes

### LINKED ISSUE #3277: [Perf] Not all Grafana boards from Meshery Adapter for Istio loading
#### Description
The "Istio Service Dashboard" board doesn't show any selectable panels.

#### Screenshots/Logs
Screen Shot 2021-06-25 at 11 44 40 AM

#### Environment:
- Meshery Version: v0.5.26
- Host OS: Mac
- Browser: Safari

---

**Contributor Resources:**
- (Contributing Docs](https://docs.meshery.io/project/contributing)

### PR DESCRIPTION
**Pull request by dhruv0000**
Signed-off-by: dhruv0000 <patel.4@iitj.ac.in>

**Description**
- Upgraded grafana SDK
- Fixed #3277 
- Updated UI to follow new Rest APIs

**Notes for Reviewers**

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: models/grafana_helper.go [hunk 1]
```diff
@@ -57,7 +57,11 @@ func (g *GrafanaClient) Validate(ctx context.Context, BaseURL, APIKey string) er
...
 		BaseURL = strings.Trim(BaseURL, "/")
 	}
-	c := sdk.NewClient(BaseURL, APIKey, g.httpClient)
+	c, err := sdk.NewClient(BaseURL, APIKey, g.httpClient)
+	if err != nil {
+		return err
+	}
+
 	if _, err := c.GetActualOrg(ctx); err != nil {
 		err = errors.Wrapf(err, "connection to grafana failed")
```
Comment (leecalcote) [inline models/grafana_helper.go:66]: `errors.New()` instead?

File: models/grafana_helper.go [hunk 2]
```diff
@@ -203,6 +215,21 @@ func (g *GrafanaClient) ProcessBoard(ctx context.Context, c *sdk.Client, board *
 					}
 				}
 				grafBoard.Panels = append(grafBoard.Panels, p1)
+			} else if p1.OfType != sdk.TextType && p1.OfType != sdk.TableType && p1.Type == "row" && len(p1.Panels) > 0 {
+				for _, p2 := range p1.Panels {
```
Comment (leecalcote) [inline models/grafana_helper.go:219]: A couple more comments through this set of nested for loops / if conditions might be nice.

**Comment by leecalcote** [comment]
Were the REST endpoints previously updated? How are we coinciding the two changes, so that no individual build is broken in this area (or are we already in this mess)?

**Comment by dhruv0000** [comment]
> Were the REST endpoints previously updated? How are we coinciding the two changes, so that no individual build is broken in this area (or are we already in this mess - already have api endpoints that were changed without updating these references)?

This was an extreme case. I guess the PR adding the endpoints was merged while the `Open API Docs` PR was open and reviewed. Both merged in the same 2-3 days.

A good strategy might be to keep the `Open API Docs` PRs open for less amount of time.

File: models/grafana_helper.go (hunk 1 above)
Comment (dhruv0000) [reply models/grafana_helper.go:66]: noted.

**Comment by leecalcote** [comment]
Good work here, @dhruv0000 👍

**Pull request merged**